### PR TITLE
Add core.coreNumber field, used to differentiate multiple cores/clusters

### PR DESCRIPTION
**Pull request by ncabatoff**
when running tests.  This is not used or exposed in prod.

Remove some test-specific code from the cluster-building helpers. The
corresponding additions go on the ent side.

### CODE REVIEW

File: helper/testhelpers/testhelpers.go [hunk 1]
```diff
@@ -265,10 +265,6 @@ func ConfClusterAndCore(t testing.T, conf *vault.CoreConfig, opts *vault.TestClu
 		"approle":  credAppRole.Factory,
 		"userpass": credUserpass.Factory,
 	}
-	coreConfig.LogicalBackends = map[string]logical.Factory{
```
Comment (calvn) [inline helper/testhelpers/testhelpers.go]: Instead of completely removing, maybe do a nil check and set these as the logical backends if none are provided.

File: helper/testhelpers/testhelpers.go [hunk 2]
```diff
@@ -305,20 +294,30 @@ func GetPerfReplicatedClusters(t testing.T, conf *vault.CoreConfig, opts *vault.
...
+	localopts := *opts
+	localopts.Logger = logger.Named("perf-pri")
+	ret.PerfPrimaryCluster, _ = ConfClusterAndCore(t, conf, &localopts)
 
-	opts2 := *opts
-	opts1.Logger = logger.Named("perf-sec")
-	ret.PerfSecondaryCluster, _ = ConfClusterAndCore(t, conf, &opts2)
+	localopts.Logger = logger.Named("perf-sec")
+	numCores := opts.NumCores
+	if numCores == 0 {
```
Comment (calvn) [inline helper/testhelpers/testhelpers.go]: Shouldn't this be called before https://github.com/hashicorp/vault/pull/6855/files#diff-56dc57f8861cd469610d671d266b7a47R299 (`ret.PerfPrimaryCluster, _ = ConfClusterAndCore(t, conf, &localopts)`)

File: helper/testhelpers/testhelpers.go (hunk 1 above)
Comment (ncabatoff) [reply helper/testhelpers/testhelpers.go]: They're completely specific to two tests (one each).  There's no reason to provide them outside of those.

File: helper/testhelpers/testhelpers.go (hunk 2 above)
Comment (ncabatoff) [reply helper/testhelpers/testhelpers.go]: I'm not sure what you're saying, can you rephrase / expand?

Comment (calvn) [reply helper/testhelpers/testhelpers.go]: Shouldn't the ordering be similar to https://github.com/hashicorp/vault/pull/6855/files#diff-56dc57f8861cd469610d671d266b7a47R330, i.e. it should go before the first `ConfClusterAndCore` call.

File: helper/testhelpers/testhelpers.go (hunk 1 above)
Comment (calvn) [reply helper/testhelpers/testhelpers.go]: Ahh, I see.

File: helper/testhelpers/testhelpers.go (hunk 2 above)
Comment (ncabatoff) [reply helper/testhelpers/testhelpers.go]: We don't need to know how many cores are used for the first cluster, as it simply assigns them sequentially.  We need to know for subsequent clusters so we can properly increment firstCoreNumber.
